experimental/delegate/mlp_aie_bf16_plugin.cpp

```diff
@@ -84,15 +87,65 @@
   #define CONVERSION_DEBUG(turnOn_) ;
 #endif
 
-// Fake bfloat16 type (assuming no C++ 23)
-using bfloat16_t = std::uint16_t;
+#ifdef ENABLE_TRACE_DELEGATE
+#define TRACE_DELEGATE(str_)                                      \
+  do {                                                            \
```

Comment:
What's the difference between `do {foo();} while (0)` and `foo();` ?

Reply:
That's an old standard trick to make a macro behave like a single statement.  Probably in this case, it doesn't matter since the body of the macro is a single statement already, but it was commonly used in general for cases like this:

```
#define BAD_MACRO(arg) \
    line1(arg); \
    line2

#define GOOD_MACRO(arg) \
    do { \
        line1(arg); \
        line2; \
    } while (0)

void func() {
    if (condition)
        BAD_MACRO(arg);
    else
        GOOD_MACRO(arg);
}
```

BAD_MACRO would cause problems, whereas GOOD_MACRO wouldn't.  Someone who wraps every block in curly braces might say that's what you get for not wrapping every block in curly braces!  :-D

I'm having second thoughts about it, though.  I think I'll go back and simplify it.

Reply:
Yikes. Macros are... quite something!

Reply:
Simplified.